repository/src/main/java/tech/ydb/yoj/repository/db/TableHelper.java

```diff
@@ -0,0 +1,68 @@
+package tech.ydb.yoj.repository.db;
+
+import com.google.common.collect.Sets;
+import tech.ydb.yoj.repository.db.cache.FirstLevelCache;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toSet;
+
+public final class TableHelper {
```

Comment:
I'd name this class something unappetizing, like `TableQueryImpl` or something, so that people don't try to reuse the logic in their own application code